libraries/wallet/wallet.cpp

```diff
@@ -1969,9 +2005,14 @@ class wallet_api_impl
       fc::optional<vesting_balance_id_type> vbid = maybe_id<vesting_balance_id_type>(witness_name);
       if( !vbid )
       {
-         witness_object wit = get_witness( witness_name );
-         FC_ASSERT( wit.pay_vb );
-         vbid = wit.pay_vb;
+         if (is_witness(witness_name)) 
+         {
+            witness_object wit = get_witness( witness_name );
+            FC_ASSERT( wit.pay_vb, "Account ${account} has no core TOKEN vested and thus its not allowed to withdraw.", ("account", witness_name));
+            vbid = wit.pay_vb;
+         }
+         else 
+            FC_THROW("Account ${account} has no core TOKEN vested and thus its not allowed to withdraw.", ("account", witness_name));
```

Comment:
If we want to throw here shouldn't we do it with an "account is not a witness" message?

Reply:
Right but here we are interested in vesting balance only. As per @bobinson 's comment on Jira ticket, We need to throw Assert message if the user doesn't have vesting balance. User can be witness, so we are checking `wit.pay_vb` too.

Reply:
got it.